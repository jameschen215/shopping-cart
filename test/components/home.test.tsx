import { type ReactNode } from "react";

import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import HomePage from "@/features/home/home";

// Mock the shadcn/ui components
vi.mock("@/components/typography", () => ({
  TypographyH1: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) => (
    <h1 className={className} data-testid="main-heading">
      {children}
    </h1>
  ),
}));

vi.mock("@/components/ui/button", () => ({
  Button: ({
    className,
    children,
    ...props
  }: {
    className?: string;
    children: ReactNode;
  }) => (
    <button className={className} data-testid="shop-now-button" {...props}>
      {children}
    </button>
  ),
}));

vi.mock("@/components/carousel-shad/carousel-shad", () => ({
  default: ({ className }: { className?: string }) => (
    <div className={className} data-testid="carousel">
      Carousel Component
    </div>
  ),
}));

describe("HomePage", () => {
  it("renders the main heading with correct text", () => {
    render(<HomePage />);

    const heading = screen.getByTestId("main-heading");
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(/shop smarter.*live better/i); // exclude span
  });

  it("renders carousel component", () => {
    render(<HomePage />);

    const carousel = screen.getByTestId("carousel");
    expect(carousel).toBeInTheDocument();
  });

  it("renders shop now button with correct text", () => {
    render(<HomePage />);

    const button = screen.getByTestId("shop-now-button");
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent("Shop Now");
  });
});

describe("HomePage Snapshots", () => {
  it("matches snapshots", () => {
    const { container } = render(<HomePage />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
