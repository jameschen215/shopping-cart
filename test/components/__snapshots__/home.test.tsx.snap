// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage Snapshots > matches snapshots 1`] = `
<div
  class="flex flex-1 flex-col items-center px-2"
>
  <h1
    class="font-display mt-15 flex flex-wrap justify-center gap-2 p-2"
    data-testid="main-heading"
  >
    <span>
      Shop Smarter,
    </span>
    <span>
      Live Better
    </span>
  </h1>
  <div
    class="mt-8"
    data-testid="carousel"
  >
    Carousel Component
  </div>
  <button
    class="mt-8 w-full max-w-xs cursor-pointer"
    data-testid="shop-now-button"
  >
    Shop Now
  </button>
</div>
`;
